validator/client/beacon-api/activation.go

```diff
@@ -0,0 +1,184 @@
+//go:build use_beacon_api
+// +build use_beacon_api
+
+package beacon_api
+
+import (
+	"encoding/json"
+	"net/http"
+	neturl "net/url"
+	"strconv"
+	"time"
+
+	"context"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/api/gateway/apimiddleware"
+	rcpmiddleware "github.com/prysmaticlabs/prysm/v3/beacon-chain/rpc/apimiddleware"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"google.golang.org/grpc"
+)
+
+func (c beaconApiValidatorClient) waitForActivation(ctx context.Context, in *ethpb.ValidatorActivationRequest) (ethpb.BeaconNodeValidator_WaitForActivationClient, error) {
+	return &beaconNodeValidatorWaitForActivationClient{
+		ctx:                        ctx,
+		beaconApiValidatorClient:   c,
+		ValidatorActivationRequest: in,
+	}, nil
+}
+
+type beaconNodeValidatorWaitForActivationClient struct {
```

Comment:
Can we make the naming less verbose here ?

Reply:
Renamed into `waitForActivationClient`